provider/core/src/fxhash.rs

```diff
@@ -0,0 +1,118 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+/// Computes the FxHash of a byte array.
+///
+/// ICU4X components can use this for a stable, compile-time hash.
+///
+/// # Examples
+///
+/// ```
+/// use zerovec::zeroslice;
+/// use zerovec::ule::AsULE;
+///
+/// let nums = zeroslice!(u32; <u32 as AsULE>::ULE::from_unsigned; [1, 2, 3, 4, 5]);
+/// let hashed = icu_provider::fxhash_32(nums.as_bytes());
+///
+/// assert_eq!(hashed, 0xF7495CFD);
+/// ```
+pub const fn fxhash_32(bytes: &[u8]) -> u32 {
+    fxhash_32_trim(bytes, 0, 0)
+}
+
+/// Const function to compute the FxHash of a byte array.
+///
+/// FxHash is a speedy hash algorithm used within rustc. The algorithm is satisfactory for our
+/// use case since the strings being hashed originate from a trusted source (the ICU4X
+/// components), and the hashes are computed at compile time, so we can check for collisions.
+///
+/// We could have considered a SHA or other cryptographic hash function. However, we are using
+/// FxHash because:
+///
+/// 1. There is precedent for this algorithm in Rust
+/// 2. The algorithm is easy to implement as a const function
+/// 3. The amount of code is small enough that we can reasonably keep the algorithm in-tree
+/// 4. FxHash is designed to output 32-bit or 64-bit values, whereas SHA outputs more bits,
+///    such that truncation would be required in order to fit into a u32, partially reducing
+///    the benefit of a cryptographically secure algorithm
+// The indexing operations in this function have been reviewed in detail and won't panic.
+#[allow(clippy::indexing_slicing)]
+pub(crate) const fn fxhash_32_trim(
```

Comment:
nit: this used to be called `_manual_slice` and that's also what the const fns in locid are called.

Reply:
we should also document this function's behavior

I think the comments about why fxhash should go on the module or on the other function, this function should have short documentation saying something like "runs `fxhash_32` with a manual slice" and then document what it means for it to be a manual slice

Reply:
1. It's not manual slice; it takes "skip start" and "skip end" instead of "start" and "limit". This function has been around for a while and this PR does not seek to change its call sites
2. This function is not public; I just moved it. I made a new public API for it.

Reply:
ah, can you still move the docs?